unidist/core/backends/mpi/core/serialization.py

```diff
@@ -56,7 +53,9 @@ def is_pickle5_serializable(data):
     bool
         ``True`` if the data should be serialized with pickle using protocol 5 (out-of-band data).
     """
-    return isinstance(data, pd.DataFrame)
+    return hasattr(data, "__reduce_ex__") and inspect.ismethod(
+        getattr(data, "__reduce_ex__")
```

Comment:
Why do we need this additional check `getattr(data, "__reduce_ex__")`?

Reply:
Just a regular Python check that it is really a method, not an attribute and it is also callable.

Reply:
`__reduce_ex__` itself is a method given by Python standard [link](https://docs.python.org/3/library/pickle.html#object.__reduce_ex__). I am wondering why you think the first check is not enough?

Reply:
You cannot be sure that it is a method, it ca be any non callable attribute in Python. It`s better to double check, instead of handling exceptions.

Reply:
The Python specification I mentioned guarantees that. I do not quite get you. If someone wants to support pickle serialization, he must define the method but not a different class member.